src/operator/tensor/elemwise_unary_op.cc

```diff
@@ -0,0 +1,21 @@
+/*!
+ *  Copyright (c) 2015 by Contributors
+ * \file elementwise_unary_op.cc
+ * \brief CPU Implementation of unary function.
+ */
+// this will be invoked by gcc and compile CPU version
+#include "./elemwise_unary_op-inl.h"
+
+namespace mxnet {
+namespace op {
+NNVM_REGISTER_OP(abs)
+.MXNET_DESCRIBE("Take absolute value of the src")
+.set_num_inputs(1)
+.set_num_outputs(1)
+.attr<nnvm::FInferShape>("FInferShape", ElemwiseShape<1,1>)
+.attr<nnvm::FInferType>("FInferType", ElemwiseType<1,1>)
+.attr<nnvm::FGradient>("FGradient", UnaryGradUseIn<"sign">)
```

Comment:
This doesn't work because string literal cannot be used as template arguments. There are a lot of ways define grad here by none as clear and simple as this.
Any suggestions?